app/src/main/java/me/quadphase/qpdex/AdvancedSearch.java

```diff
@@ -0,0 +1,508 @@
+        if(lowerLimit > 0 || upperLimit < 999){
+            ArrayList<Integer> tempFilterPokemonArrayList = new ArrayList<Integer>();
+            tempFilterPokemonArrayList = pokemonFactory.getAllUniqueIDsFromStat(lowerLimit, upperLimit, statName);
+            currentFilterPokemonArrayList.retainAll(tempFilterPokemonArrayList);
+            tempFilterPokemonArrayList.clear();
+        }
+
+    }
+
+
+
+
+
+
+    public void switchToPokedex(View view){
+        long startTime = System.nanoTime();
+        Intent intent = new Intent(this,PokedexActivity.class);
+
+        // Obtain all stats chosen
+        collectStats();
+
+        //Comment this out, takes way tooo long 4+sec
+        currentFilterPokemonArrayList.clear();
+        currentFilterPokemonArrayList.addAll(filteredUniqueIds);
+
+        ArrayList<Integer> tempFilterPokemonArrayList = new ArrayList<>();
+
+        // Filtering algorithm, will go through a list of all uniqueIDs and apply filters
+        // consecutively. Once the list is filtered, convert the unique to nationalIds
+
+        //Start with the ability chosen, if any. Use the ability ID to get all NationalID's associated
+        if(selectedAbility.getName() != "None"){
+
+            currentFilterPokemonArrayList = pokemonFactory.getAllUniqueIDsFromAbility(Arrays.asList(allAbilities).indexOf(selectedAbility));
+        }
+
+
+        // Filter the remaining IDs by type twice
+
+        if(selectedType1.getTypeID() != 0){
+
+            tempFilterPokemonArrayList = pokemonFactory.getAllUniqueIDsFromType(Arrays.asList(allTypes).indexOf(selectedType1));
+            currentFilterPokemonArrayList.retainAll(tempFilterPokemonArrayList);
+            tempFilterPokemonArrayList.clear();
+        }
+
+
+
+        if(selectedType2.getTypeID() != 0){
+
+            tempFilterPokemonArrayList = pokemonFactory.getAllUniqueIDsFromType(Arrays.asList(allTypes).indexOf(selectedType2));
+            currentFilterPokemonArrayList.retainAll(tempFilterPokemonArrayList);
+            tempFilterPokemonArrayList.clear();
+        }
+
+        // If at least one is not default value, filter
+        filterStats(selectedHpLower, selectedHpGreater, "HP");
+        filterStats(selectedAttLower, selectedAttGreater, "ATTACK");
+        filterStats(selectedDefLower, selectedDefGreater, "DEFENCE");
+        filterStats(selectedSpattLower, selectedSpattGreater, "SPATTACK");
+        filterStats(selectedSpdefLower, selectedSpdefGreater, "SPDEFENCE");
+        filterStats(selectedSpeedLower, selectedSpeedGreater, "SPEED");
+        //TODO: Filter by sum
+        //filterStats(selectedSpeedLower, selectedSpeedGreater, "SUM");
+
+        // When Done filtering by UniqueID's convert uniqueIDs to NationalId's
+        filteredNationalIds = pokemonFactory.convertUniqueToNational(currentFilterPokemonArrayList);
+
+        // Filter national IDs based on Egg Groups
+        if(selectedEggGroup1.getName() != "None"){

```

Comment:
Should be: 
if (!selectedEggGroup1.getName().equals("None")) {
With strings in Java, use blabla.equals("") instead of == or !blabla.equals("") instead of !=

Same for selectedEggGroup2 below